File: src/courses/data/catalog/02-dsa/dsa-core/implementations/csharp/linked-lists.cs
/**
 * C# Implementation: Linked Lists
 * Invariant: Insert/Delete O(1) given pointer, Search O(N)
 */
using System;
using System.Collections.Generic;

namespace Seed.DSA.Core
{
    public class LinkedLists<T>
    {
        private readonly List<T> _items = new List<T>();

        // TODO: Implement algorithm operations
        public void Add(T item)
        {
            _items.Add(item);
        }

        public int Count => _items.Count;
    }
}

File: src/courses/data/catalog/02-dsa/dsa-core/implementations/c/linked-lists.c
/*
 * C Implementation: Linked Lists
 * Invariant: Insert/Delete O(1) given pointer, Search O(N)
 */
#include <stdio.h>
#include <stdlib.h>

typedef struct {
    int* data;
    int size;
    int capacity;
} linked_lists_t;

// TODO: Complete the allocation and operations
void init_linked_lists(linked_lists_t* item, int capacity) {
    item->data = (int*)malloc(capacity * sizeof(int));
    item->size = 0;
    item->capacity = capacity;
}
